docs/design/deviceshifu/design-deviceShifu-rtsp.md

```diff
@@ -0,0 +1,122 @@
+# deviceShifu RTSP overall design
+
+deviceShifu-RTSP allows shifu connect to rtsp server and save the video into file. User can watch rtsp video using media player like VLC and get history video file from PV.
+
+## Goal
+
+Create a deviceShifu-RTSP using third-party library and allow user to connect RTSP server, and record the rtsp stream into file like mp4 / flv and so on, user can export file out of k8s.
+
+## General Design
+
+deviceShifu-RTSP will receive RESTful style request like other, then transfer the request into RTSP method calls and send it to RTSP Server, meanwhile deviceShifu-RTSP provide a RTSP address which forward the RTSP stream from the target RTSP server.
+
+## Detailed Design
+
+### Protocol Specification
+
+deviceShifu-RTSP provide a RTSP server to forward RTSP stream and HTTP Server to handle Request.
+
+```go
+type ProtocolSettings struct {
+    MQTTSetting *MQTTSetting `json:"MQTTSetting,omitempty"`
+    ....
++   RTSPSetting *RTSPSetting `json:"RTSPSetting,omitempty"`
+}
+
+type RTSPSetting struct {
+    Username        string `json:"username,omitempty"`
+    PasswordSecret  string `json:"passwordSecret,omitempty"`
+}
+```
+
+If user turn on Record, deviceShifu will using telemetry Service to push the rtsp stream to telemetry Service, and telemetry Service will forward the stream to the rtsp recording server to record the rtsp video into file.
```

Comment:
1. Can you please explain more on how user can turn on Record, how does the command propagate from deviceShifu to TelemetryService, then RTSP server?
2. After the user turn on record, deviceShifu will actively forward the TCP traffic to the TelemetryService, and RTSP server will start the recording. This means RTSP server is reading username/password from K8s client?

Reply:
for the rtsp has the redirect function, I think we needn't to think those